vnext/PropertySheets/ReactPatches.props

```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- 
+ Copyright (c) Microsoft Corporation. All rights reserved.
+ Licensed under the MIT License.. 
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!--
+    Visual Studio has its own incremental build logic for simple cases that
+    falls over for patching. Force use of MSBuild logic.
+  -->
+  <PropertyGroup>
+    <DisableFastUpToDateCheck>True</DisableFastUpToDateCheck>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <!--
+      Do not patch facebook/react-native by default until we make the switch
+      off of microsoft/react-native.
+    -->
+    <PATCH_RN Condition="'$(OSS_RN)'!='true'">true</PATCH_RN>
+
+    <ReactNativeDir Condition="'$(BaseIntDir)'!=''">$(BaseIntDir)\PatchedReactNative\react-native</ReactNativeDir>
+    <ReactNativeDir Condition="'$(BaseIntDir)'==''">$(BaseIntermediateOutputPath)\$(Platform)\$(Configuration)\PatchedReactNative\react-native</ReactNativeDir>
```

Comment:
This one intentionally includes platform and config where the above doesn't. I think we should maybe just override BaseIntermediateOutputPath with BaseIntDir if we have it, but most of the world seems to refer to IntDir right now. Possibly because (not base) IntermediateOutputDirectory is a dynamic property which makes it a pain to use.